catroid/src/main/java/org/catrobat/catroid/ui/fragment/FormulaEditorFragment.java

```diff
@@ -418,7 +419,13 @@ public void setValue(int value) {
 
 			@Override
 			public int getValue() {
-				String currentValue = getSelectedFormulaText();
+				String currentValue;
+				if (getSelectedFormulaText() != null) {
+					currentValue = getSelectedFormulaText().toUpperCase(Locale.US);
+				} else {
+					currentValue = null;
+				}
+
 				if (currentValue != null && currentValue.matches("^#[0-9A-F]{6}$")) {
```

Comment:
This should be enough to fix the issue:
```
if (currentValue != null && currentValue.matches("^#[0-9A-Fa-f]{6}$")) {
```

Reply:
Ah, Right! I thought only upper case input can invoke colour, thanks for the suggestion!